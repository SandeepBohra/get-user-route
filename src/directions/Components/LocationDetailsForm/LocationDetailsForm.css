.LocationDetailsForm input {
    width: 90%;
    padding: 8px;
    border: 1px solid black;
}

.LocationDetailsForm p {
    font-size: 16px;
    font-weight: 500;
}

.form-buttons button {
    padding: 10px;
    font-size: 16px;
    margin-top: 50px;
}

.route-details {
    margin-top: 25px;
    font-size: 16px;
    font-weight: 500;
}