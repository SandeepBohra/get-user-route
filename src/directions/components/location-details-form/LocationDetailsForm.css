.location-details-form input {
    width: 90%;
    padding: 8px;
    border: 1px solid black;
}

.location-details-form p {
    font-size: 16px;
    font-weight: 500;
}

.form-buttons button {
    padding: 10px;
    font-size: 16px;
    margin-top: 50px;
}

.route-details {
    margin-top: 25px;
    font-size: 16px;
    font-weight: 500;
}