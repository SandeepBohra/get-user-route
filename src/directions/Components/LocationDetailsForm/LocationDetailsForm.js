import React, { Component } from 'react';
import { googleMaps } from '../../Services/googleMap'
import { DirectionInfo } from '../DirectionInfo/DirectionInfo'
import { GooglePlacesAutocomplete } from '../../../Shared/Components/index'
import './LocationDetailsForm.css'


class LocationDetailsForm extends Component {
    originInput;
    originInputAutoComplete;
    destInputAutoComplete;
    destInput;
    constructor() {
        super()
        this.state = {
            submitEnabled: false,
            startingLocation: '',
            dropOffPoint: '',
        }
    }

    renderAutoComplete = async () => {
        const maps = await this.props.googleMaps();

        this.originInputAutoComplete = new maps.places.Autocomplete(this.originInput);
        this.destInputAutoComplete = new maps.places.Autocomplete(this.destInput);
    };

    componentDidMount() {
        this.renderAutoComplete();
    }

    handleSubmit = () => {
        const startingPoint = this.originInputAutoComplete.getPlace();
        const dropPoint = this.destInputAutoComplete.getPlace();
        this.props.sendLocationAndGetRoute(startingPoint, dropPoint)
    }

   enableSubmit = () => {
       const startingLocation = this.state.startingLocation
       const dropOffPoint = this.state.dropOffPoint
        return startingLocation && dropOffPoint
    }

    handleChange = e => {
        debugger
        const {name, value} = e.target
        this.setState({
            [name]: value,
        })
    }

    clearInputFields = () => {
        this.setState({
            startingLocation: '',
            dropOffPoint: '',
        })
        this.originInput.value = '';
        this.destInput.value = '';
    }

    resetLocationDetails = () => {
        this.clearInputFields();
        this.props.resetExistingRouteInformation()
    }

    render() {
        const {routeDistance, routeTime} = this.props
        return (
            <div className="LocationDetailsForm">
                <GooglePlacesAutocomplete 
                    label="Strarting location"
                    handleInputChange={this.handleChange}
                    fieldRef={elem => (this.originInput = elem)}
                    inputName="startingLocation"
                />
                <GooglePlacesAutocomplete 
                    label="Drop-off point"
                    handleInputChange={this.handleChange}
                    fieldRef={elem => (this.destInput = elem)}
                    inputName="dropOffPoint"
                />
                {this.props.showRouteDistAndTime ?
                <DirectionInfo 
                    totalDistance={routeDistance}
                    totalTime={routeTime}
                />
                : null}
                <div className="form-buttons">
                    <button 
                        disabled={!this.enableSubmit()} 
                        style={{marginRight: `30%`}} 
                        onClick={this.handleSubmit}>Submit</button>
                    <button 
                        onClick={this.resetLocationDetails}>Reset</button>
                </div>
            </div>
        );
    }
}

export default LocationDetailsForm;

LocationDetailsForm.defaultProps = {
    googleMaps,
}