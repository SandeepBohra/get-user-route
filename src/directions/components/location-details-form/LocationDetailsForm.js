import React, { Component } from 'react';
import { googleMaps } from '../../services/google-maps/googleMap';
import DirectionInfo from '../direction-info/DirectionInfo';
import { InputPlacesAutocomplete } from '../../../shared/components/index';
import './LocationDetailsForm.css';


// This component is used to get the start and drop off location from the user and also to submit the locations to
// the API to get the path details
class LocationDetailsForm extends Component {
    constructor() {
        super();
        this.state = {
            startingLocation: '',
            dropOffPoint: '',
        }
    }

    handleSubmit = () => {
        const { startingLocation, dropOffPoint } = this.state;
        this.props.sendLocationAndGetRoute(startingLocation, dropOffPoint);
    }

    // Below method enables or disables the submit button
    enableSubmit = () => {
       const { startingLocation, dropOffPoint } = this.state;
        return startingLocation && dropOffPoint;
    }

    // Method to enable or disable button if input fields have some value
    enableReset = () => {
        const { startingLocation, dropOffPoint } = this.state;
        return startingLocation || dropOffPoint;
    }

    // update states(startingLocation and DropOffPoint) on selecting the places from autocomplete
    handleOnSelectAddress = (place, name) => {
        this.setState({
            [name]: place,
        })
    }

    // clear the values from form input fields
    clearInputFields = () => {
        this.setState({
            startingLocation: '',
            dropOffPoint: '',
        })
    }

    // Below method is used to reset the form inputs as well as reset the route on Map
    resetLocationDetails = () => {
        this.clearInputFields();
        this.props.resetExistingRouteInformation();
    }

    render() {
        const { routeDetails } = this.props;
        return (
            <div className="location-details-form">
                <InputPlacesAutocomplete 
                    label="Starting location"
                    handleOnSelectAddress={this.handleOnSelectAddress}
                    location={this.state.startingLocation}
                    inputName="startingLocation"
                />
                <InputPlacesAutocomplete 
                    label="Drop-off point"
                    handleOnSelectAddress={this.handleOnSelectAddress}
                    location={this.state.dropOffPoint}
                    inputName="dropOffPoint"
                />
                {
                  this.props.showRouteDistAndTime 
                  ? <DirectionInfo 
                      totalDistance={routeDetails.routeDistance}
                      totalTime={routeDetails.routeTime} />
                  : null
                }
                <div className="form-buttons">
                    <button 
                        disabled={!this.enableSubmit()} 
                        style={{marginRight: `30%`}} 
                        onClick={this.handleSubmit}>Submit</button>
                    <button 
                        disabled={!this.enableReset()}
                        onClick={this.resetLocationDetails}>Reset</button>
                </div>
            </div>
        );
    }
}

export default LocationDetailsForm;

LocationDetailsForm.defaultProps = {
    googleMaps,
}