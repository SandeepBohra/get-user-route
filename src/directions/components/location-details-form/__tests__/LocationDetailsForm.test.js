import React from 'react';
import renderer from 'react-test-renderer';
import LocationDetailsForm from '../LocationDetailsForm';

describe('<LocationDetailsForm /> component', () => {
  
  it('should match the snapshot', () => {
    const output = renderer.create(<LocationDetailsForm />).toJSON();
    expect(output).toMatchSnapshot();
  })
})