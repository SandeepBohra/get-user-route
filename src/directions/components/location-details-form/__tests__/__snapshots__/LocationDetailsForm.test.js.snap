// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocationDetailsForm /> component should match the snapshot 1`] = `
<div
  className="location-details-form"
>
  <div
    className="input-places-autocomplete"
  >
    <p>
      Strarting location
    </p>
    <input
      name="startingLocation"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="input-places-autocomplete"
  >
    <p>
      Drop-off point
    </p>
    <input
      name="dropOffPoint"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="form-buttons"
  >
    <button
      disabled={true}
      onClick={[Function]}
      style={
        Object {
          "marginRight": "30%",
        }
      }
    >
      Submit
    </button>
    <button
      onClick={[Function]}
    >
      Reset
    </button>
  </div>
</div>
`;
